src/main/java/com/google/ehub/data/FavoriteItemDatastore.java

```diff
@@ -55,6 +61,8 @@ public void addFavoriteItem(String userEmail, Long itemId) {
     favoriteItemEntity.setProperty(ITEM_ID_PROPERTY_KEY, itemId);
 
     datastoreService.put(favoriteItemEntity);
+
+    addLikeToItem(itemId);
```

Comment:
This is another place where we would use database transactions.
Transactions are atomic (all or nothing). That means if there is a failure in adding the FavoriteItemEntity to the database, then we won't add the like to the EntertainmentItemDatastore.

Without a transaction, we could have a failure in one query but not the other, which would lead to inconsistencies.